src/settings.rs

```diff
@@ -2,18 +2,97 @@ use crate::LOG_DRAIN;
 
 use serde::{Deserialize, Serialize};
 use slog::info;
+use std::collections::HashSet;
 
+#[derive(Serialize, Deserialize, Default, Debug)]
+#[serde(default)]
+pub(crate) struct IDRange {
+    pub min: i64,
+    pub max: i64,
+}
+
+#[derive(Serialize, Deserialize, Default, Debug)]
+#[serde(default)]
+pub(crate) struct RuleStrategy {
+    pub rule: String,
+    pub ranges: Vec<IDRange>,
+}
+
+fn valid_ranges(ranges: &[IDRange], prefix_error_message: &str) -> Result<(), String> {
+    for range in ranges {
+        if range.min > range.max {
```

Comment:
I would also check that the numbers are valid ones for uid's and gid's: one step would be to change i64 to u64 of IDRange -- so you rule out negative numbers, but also check that they are valid uid's. Reference: https://systemd.io/UIDS-GIDS/, because not all positive numbers apply.